lib/galaxy/jobs/__init__.py

```diff
@@ -1770,10 +1777,14 @@ def reclaim_ownership( self ):
     def user_system_pwent( self ):
         if self.__user_system_pwent is None:
             job = self.get_job()
-            try:
-                self.__user_system_pwent = pwd.getpwnam( job.user.email.split('@')[0] )
-            except:
-                pass
+
+            # try to get user from email address/username
+            for cand in [job.user.email.split('@')[0], job.user.username]:
```

Comment:
I'd prefer this to be configurable via a variable in galaxy.ini (sth. like `real_system_username`, with options 'user_email' and 'username'), close to the `external_chown_script`.